Fix up hg support in `jest-changed-files`.

cpojer opened the pull request.

cpojer commented:
@facebook-github-bot shipit

facebook-github-bot-9 commented:
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/1082266645169611/int_phab) to review.

dmitriiabramov commented on packages/jest-changed-files/src/__tests__/hg-test.js:
```diff
@@ -32,13 +32,13 @@ describe('hgChecker', () => {
-    pit('returns null for non hg repo folder', () => {
-      return hg.isHGRepository(tmpdir).then(res => {
+    it('returns null for non hg repo folder', () =>
+      hg.isHGRepository(tmpdir).then(res => {
```
but you're not returning anything... that will always pass

cpojer replied:
yes we are, it's an arrow function:

`it('tests a promise', () => foo.then(…))`

dmitriiabramov commented on packages/jest-changed-files/src/__tests__/hg-test.js:
```diff
@@ -53,13 +53,13 @@ describe('hgChecker', () => {
-    pit('returns an empty array for hg repo folder without modified files', () => {
-      return hg.findChangedFiles(tmpdir).then(res => {
+    it('returns an empty array for hg repo folder without modified files', () =>
+      hg.findChangedFiles(tmpdir).then(res => {
```
same

cpojer commented:
I should explain why this change was necessary:

`HGPLAIN=1 hg status` works on www and works in open source mercurial. However, it doesn't work on Facebook laptops because they require the rest of the env variables to be available, otherwise the test will throw and this feature won't work. I changed it so HGPLAIN is added to all env variables that are available.

I also added an error handler, which should be common sense but I think we didn't have one there because the code would never be called if hg (or git) was not available based on the previous call. Anyway, error handlers are good for you. Use them!
